google-cloud-logging/lib/google/cloud/logging/v2/logging_service_v2_api.rb

```diff
@@ -299,13 +300,7 @@ def write_log_entries \
             )
             req.log_name = log_name unless log_name.nil?
             req.resource = resource unless resource.nil?
-            # Custom code here to work around setting a Hash where a Map is expected.
```

Comment:
As @jmuk suggested, this should be fixed in 'toolkit'. In the mean time, let me add your workaround back.